Source/Core/Common/Debug/CodeTrace.cpp

```diff
@@ -0,0 +1,394 @@
+    size = 2;
+  else if (op.find(double_word) != std::string::npos || op.find(paired) != std::string::npos)
+    size = 8;
+
+  return size;
+}
+
+const std::tuple<std::vector<std::string>, std::set<u32>> CodeTrace::GetTracked()
+{
+  // Output without conversions, incase the u32 memory address is to be used.
+  return std::make_tuple(m_reg_autotrack, m_mem_autotrack);
+}
+
+void CodeTrace::SetRegTracked(const std::string reg)
+{
+  m_reg_autotrack.push_back(reg);
+}
+
+const InstructionAttributes CodeTrace::GetInstructionAttributes(TraceOutput instruction) const
+{
+  // Slower process of breaking down saved instruction. Only used when stepping through code if a
+  // decision has to be made, otherwise used afterwards on a log file.
+  InstructionAttributes tmp_attributes;
+  tmp_attributes.instruction = instruction.instruction;
+  tmp_attributes.address = PC;
+  std::string instr = instruction.instruction;
+
+  // Convert sp, rtoc, and ps to r1, r2, and F#. ps is handled like a float operation.
+  std::regex replace_sp("(\\W)sp");
+  std::regex replace_rtoc("rtoc");
+  std::regex replace_ps("(\\W)p(\\d+)");
+  instr = std::regex_replace(instr, replace_sp, "$1r1");
+  instr = std::regex_replace(instr, replace_rtoc, "r2");
+  instr = std::regex_replace(instr, replace_ps, "$1f$2");
+
+  // Pull all register numbers out and store them. Limited to Reg0 if ps operation, as ps get
+  // too complicated to track easily.
+  std::regex regis(
+      "\\W([rfp]\\d+)[^r^f]*(?:([rf]\\d+))?[^r^f\\d]*(?:([rf]\\d+))?[^r^f\\d]*(?:([rf]\\d+))?");
+  std::smatch match;
+
+  // ex: add r4, r5, r6 -> r4 = Reg0, r5 = Reg1, r6 = Reg2. Reg0 is always the target register.
+  if (std::regex_search(instr, match, regis))
+  {
+    tmp_attributes.reg0 = match.str(1);
+    if (match[2].matched)
+      tmp_attributes.reg1 = match.str(2);
+    if (match[3].matched)
+      tmp_attributes.reg2 = match.str(3);
+    if (match[4].matched)
+      tmp_attributes.reg3 = match.str(4);
+
+    if (instruction.memory_target)
+    {
+      tmp_attributes.memory_target = instruction.memory_target;
+      tmp_attributes.memory_target_size = GetMemoryTargetSize(instr);
+
+      if (StringBeginsWith(instr, "st") || StringBeginsWith(instr, "psq_s"))
+        tmp_attributes.is_store = true;
+      else
+        tmp_attributes.is_load = true;
+    }
+  }
+
+  return tmp_attributes;
+}
+
+const TraceOutput CodeTrace::SaveInstruction() const
+{
+  // Quickly save instruction and memory target for fast logging.
+  TraceOutput output;
+  std::string instr = PowerPC::debug_interface.Disassemble(PC);
+  output.instruction = instr;
+  output.address = PC;
+
+  if (IsInstructionLoadStore(output.instruction))
+    output.memory_target = PowerPC::debug_interface.GetMemoryAddressFromInstruction(instr);
+
+  return (output);
+}
+
+bool CompareMemoryTargetToTracked(const std::string& instr, const u32& mem_target,
+                                  std::set<u32>& mem_tracked)
```

Comment:
`const u32 mem_target, std::set<u32>* mem_tracked`.  `u32` can be passed as copy, `std::set` should be passed as a pointer if it's modified, according to Dolphin coding style.

Reply:
This isn't changing anything, just using an iterator to read it. Does the rule still apply in this case?

Reply:
If the value isn't modified, you should make it a const reference: `const std::set<u32>&`.